core/ArchiveProcessor/LoaderLock.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\ArchiveProcessor;
+
+use Piwik\Archive\ArchiveInvalidator;
+use Piwik\Cache;
+use Piwik\Common;
+use Piwik\Concurrency\Lock;
+use Piwik\Concurrency\LockBackend;
+use Piwik\Config;
+use Piwik\Container\StaticContainer;
+use Piwik\Context;
+use Piwik\DataAccess\ArchiveSelector;
+use Piwik\DataAccess\ArchiveTableCreator;
+use Piwik\DataAccess\ArchiveWriter;
+use Piwik\DataAccess\Model;
+use Piwik\DataAccess\RawLogDao;
+use Piwik\Date;
+use Piwik\Db;
+use Piwik\Option;
+use Piwik\Period;
+use Piwik\Piwik;
+use Piwik\SettingsServer;
+use Piwik\Site;
+use Psr\Log\LoggerInterface;
+use Piwik\CronArchive\SegmentArchiving;
+
+/**
+ * This class uses PluginsArchiver class to trigger data aggregation and create archives.
+ */
+class LoaderLock
```

Comment:
@peterhashair can we write tests for this class to make sure it works?

Reply:
@tsteur add a test,  simple one, let me know if that doesn't meet the requirements

Reply:
@peterhashair can you move this test into a `LoaderLockTest.php`? For each class/file we also create a different test class.

What you additionally want to test is that

* Create another lock with a different idea, check that the initially created lock doesn't also lock this lock with different ID. To make sure it only locks the specific ID.
* A test where you enter a long ID with > 64 characters
* Call "unlock" when it's not locked yet
* Are there other edge cases that can be tested?

Reply:
@tsteur update it, but it seems like Travis number 7 angular test is started to fail.  [https://app.travis-ci.com/github/matomo-org/matomo/jobs/547651253](https://app.travis-ci.com/github/matomo-org/matomo/jobs/547651253). There may be a regression somewhere